src/cargo/core/compiler/build_config.rs

```diff
@@ -66,7 +66,11 @@ impl BuildConfig {
                  its environment, ignoring the `-j` parameter",
             )?;
         }
-        let jobs = jobs.or(cfg.jobs).unwrap_or(::num_cpus::get() as u32);
+        let jobs = match jobs.or(cfg.jobs) {
+            None => ::num_cpus::get() as u32,
+            Some(j) if j < 0 => (::num_cpus::get() as i32 + j).max(1) as u32,
```

Comment:
I think I'd personally prefer that this return an error if the result is less than one rather than clamping to one

Reply:
@alexcrichton I think it's better if not, suppose that a build script contains `-j -2` but is then run on a dual-core machine, now your build breaks.

Reply:
Hm this is something I'm worried about though in that it's sort of hidden behavior. Using `-j100` when you have 2 cpus just means "spawn up to 100 things at the same time" which is always possible, you're basically asking the kernel to do tons of scheduling for you. If the request ends up being "spawn at most -N" tasks I'm not sure Cargo can reasonbly handle that and I would prefer we be conservative and return an error.

If in the future, though, the script case is common enough and the error is just getting in everyone's way we could change it to clamp to 1